File: scripts/algorithms/A/Apply Operations to an Array/Apply Operations to an Array.java
// Runtime: 0 ms (Top 100.0%) | Memory: 44.10 MB (Top 19.85%)

class Solution {
    public int[] applyOperations(int[] nums) {
        int length = nums.length;
        int [] arr = new int[length];

        for (int i = 0 , j = 0; i < length; i++)
        {
            if(nums[i] == 0) continue;
            
            if (i < length-1 && nums[i] == nums[i+1])
            {
                arr[j++] = 2 * nums[i];
                nums[++i] = 0;  // or i++;  only
            }else{
                arr[j++] = nums[i];
            }
        }
        return arr;
     }
}

File: scripts/algorithms/A/Apply Operations to an Array/Apply Operations to an Array.js
// Runtime: 57 ms (Top 73.96%) | Memory: 51.20 MB (Top 8.04%)

var applyOperations = function(nums) {
// perform the operation on each array element
    for(let i = 0; i < nums.length - 1; i++){
        if(nums[i] === nums[i + 1]){
            nums[i] *= 2;
            nums[i + 1] = 0;
        }
    }
// Move zeroes to end
    let index = 0;
    nums.forEach(num => {
        if(num !== 0)
            nums[index++] = num;
    });
    while(index < nums.length){
        nums[index++] = 0;
    }
    return nums;
};
